tools/box_labels_AbID

```diff
@@ -0,0 +1,87 @@
+                
+                <fo:page-sequence-master master-name="repeatME">
+                    <fo:repeatable-page-master-reference master-reference="boxLabels"/>
+                </fo:page-sequence-master>
+            </fo:layout-master-set>
+            <xsl:apply-templates select="ead:ead"/>
+        </fo:root>
+    </xsl:template>
+    
+    <xsl:template match="ead:ead">
+        <fo:page-sequence master-reference="repeatME">
+            <fo:flow flow-name="xsl-region-body">
+                <fo:table>
+                    <fo:table-column column-width="96mm"/>
+                    <fo:table-body font-family="Arial">
+                        <xsl:apply-templates select="//ead:dsc"/>
+                    </fo:table-body>
+                </fo:table>
+            </fo:flow>
+        </fo:page-sequence>
+    </xsl:template>
+    
+    <xsl:template match="ead:container[@type='Box' or @type='box' or @type='volume' or @type='Volume']">
+        
+        <xsl:for-each select=".[not(.=preceding::ead:container[@type='Box' or @type='box'  or @type='volume' or @type='Volume'])]">
+                <xsl:apply-templates mode="single" select="."/>
+        </xsl:for-each>
+    </xsl:template>
+    
+    <xsl:template mode="single"
+        match="ead:container[@type='Box' or @type='box' or @type='volume' or @type='Volume']">
+        <fo:table-row height="3.3in" padding-bottom=".2in" padding-top=".2in" margin-left="3mm"
+            margin-right="3mm">
+            <fo:table-cell>
+                <fo:block padding-bottom=".2in" padding-top=".2in" padding-left="2mm"
+                    padding-right="2mm" border-color="black" border-style="solid"
+                    border-width="thin" margin-top="9mm">
+                    <fo:block font-weight="bold" font-size="16pt" font-family="Arial"
+                        text-align="center" span="none" padding-after="12pt">
+                        <xsl:value-of select="//ead:archdesc/ead:did/ead:unittitle"/>
+                    </fo:block>
+                	<fo:block font-weight="bold" font-size="12pt" font-family="Arial"
+                		text-align="center" span="none" padding-after="16pt" padding-before="6pt">
+                		<xsl:value-of select="//ead:eadheader/ead:eadid"/>
+                	</fo:block>
+                    <fo:block font-weight="bold" font-size="20pt" font-family="Arial"
+                        text-align="center" span="none" padding-after="12pt">
+                        &#x2028;<fo:leader/>
+                        <xsl:value-of
+                            select="concat(upper-case(substring(normalize-space(@type),1,1)),
+                            substring(normalize-space(@type), 2))"/>
+                        <xsl:text> </xsl:text>
+                        <xsl:value-of select="."/>
+                        &#x2028;<fo:leader/>
+                    </fo:block>
+                    <fo:block text-align="left" line-height=".3" vertical-align="after">
```

Comment:
I am _really_ out of my depth here, but is "after" a valid value for vertical-align?  fop complained about it when I tried to generate a PDF.

Reply:
Aha, there is a reply to the stackoverflow comment that says that Apache FOP doesn't support vertical-align="after".

So, since it's meant to run in Oxygen, and it works well in Oxygen, I think it's just fine as it is/was.